packages/connectors/src/walletConnect.ts

```diff
@@ -3,346 +3,173 @@ import {
   SwitchChainError,
   UserRejectedRequestError,
-  getClient,
   normalizeChainId,
 } from '@wagmi/core'
-import type WalletConnectProvider from '@walletconnect/ethereum-provider'
-import type {
-  UniversalProviderOpts,
-  UniversalProvider as UniversalProvider_,
-} from '@walletconnect/universal-provider'
-import type { Web3Modal } from '@web3modal/standalone'
+import type EthereumProviderType from '@walletconnect/ethereum-provider'
 import { providers } from 'ethers'
 import { getAddress, hexValue } from 'ethers/lib/utils.js'
 
 import { Connector } from './base'
 
-// Shared config for WalletConnect v2
-const defaultV2Config = {
-  namespace: 'eip155',
-  methods: [
-    'eth_sendTransaction',
-    'eth_sendRawTransaction',
-    'eth_sign',
-    'eth_signTransaction',
-    'eth_signTypedData',
-    'eth_signTypedData_v3',
-    'eth_signTypedData_v4',
-    'personal_sign',
-    'wallet_switchEthereumChain',
-    'wallet_addEthereumChain',
-  ],
-  events: ['accountsChanged', 'chainChanged'],
-}
-
-type UniversalProvider = InstanceType<typeof UniversalProvider_>
+// -- Types --------------------------------------------------------------------
+type RpcMethod =
+  | 'personal_sign'
+  | 'eth_sendTransaction'
+  | 'eth_accounts'
+  | 'eth_requestAccounts'
+  | 'eth_call'
+  | 'eth_getBalance'
+  | 'eth_sendRawTransaction'
+  | 'eth_sign'
+  | 'eth_signTransaction'
+  | 'eth_signTypedData'
+  | 'eth_signTypedData_v3'
+  | 'eth_signTypedData_v4'
+  | 'wallet_switchEthereumChain'
+  | 'wallet_addEthereumChain'
+  | 'wallet_getPermissions'
+  | 'wallet_requestPermissions'
+  | 'wallet_registerOnboarding'
+  | 'wallet_watchAsset'
+  | 'wallet_scanQRCode'
+
+type RpcEvent =
+  | 'accountsChanged'
+  | 'chainChanged'
+  | 'message'
+  | 'disconnect'
+  | 'connect'
 
 type WalletConnectOptions = {
-  /** When `true`, uses default WalletConnect QR Code modal */
+  projectId: string
   qrcode?: boolean
-} & (
-  | ({
-      version?: '1'
-    } & ConstructorParameters<typeof WalletConnectProvider>[0])
-  | ({
-      /**
-       * Project ID associated with [WalletConnect account](https://cloud.walletconnect.com)
-       */
-      projectId: NonNullable<UniversalProviderOpts['projectId']>
-      version: '2'
-    } & Omit<UniversalProviderOpts, 'projectId'>)
-)
-
-type WalletConnectSigner = providers.JsonRpcSigner
+  methods?: RpcMethod[]
+  events?: RpcEvent[]
+}
+
+type ConnectorConfig = { chains?: Chain[]; options: WalletConnectOptions }
+
+type ConnectArguments = { chainId?: number; pairingTopic?: string }
+
+type ChainIdArguments = { chainId?: number }
 
+// -- Constants ----------------------------------------------------------------
+const NAMESPACE = 'eip155'
+
+// -- Connector ----------------------------------------------------------------
 export class WalletConnectConnector extends Connector<
-  WalletConnectProvider | UniversalProvider,
+  EthereumProviderType,
   WalletConnectOptions,
-  WalletConnectSigner
+  providers.JsonRpcSigner
```

Comment:
```suggestion
  WalletConnectSigner
```

Can we keep this like before?